pennylane/transforms/get_dag_commutation.py

```diff
@@ -0,0 +1,1810 @@
+        u3 (pennylane.u3): One qubit U3 rotation.
+
+    Returns:
+         qml.operation: Simplified rotation if possible.
+    """
+
+    if (
+        np.allclose(np.mod(u3.data[0], 2 * np.pi), 0)
+        and not np.allclose(np.mod(u3.data[1], 2 * np.pi), 0)
+        and np.allclose(np.mod(u3.data[2], 2 * np.pi), 0)
+    ):
+        return qml.PhaseShift(u3.data[1], wires=u3.wires)
+    if (
+        np.allclose(np.mod(u3.data[2], 2 * np.pi), np.pi / 2)
+        and np.allclose(np.mod(u3.data[1] + u3.data[2], 2 * np.pi), 0)
+        and not np.allclose(np.mod(u3.data[0], 2 * np.pi), 0)
+    ):
+        return qml.RX(u3.data[1], wires=u3.wires)
+    if (
+        not np.allclose(np.mod(u3.data[0], 2 * np.pi), 0)
+        and np.allclose(np.mod(u3.data[1], 2 * np.pi), 0)
+        and np.allclose(np.mod(u3.data[2], 2 * np.pi), 0)
+    ):
+        return qml.RY(u3.data[0], wires=u3.wires)
+
+    return u3
+
+
+def simplify(operation):
+    r"""Simplify a rotation/ controlled rotation into RX, RY and RZ rotations.
+
+    Args:
+        operation (pennylane.Operation): Rotation or controlled rotation.
+
+    Returns:
+         qml.operation: Simplified rotation if possible.
+    """
+    if operation.name not in ["Rot", "U2", "U3", "CRot"]:
+        raise qml.QuantumFunctionError(f"{operation.name} is not a Rot, U2, U3 or CRot.")
+
+    if operation.name == "Rot":
+        return simplify_rotation(operation)
+
+    if operation.name == "U2":
+        return simplify_u2(operation)
+
+    if operation.name == "U3":
+        return simplify_u3(operation)
+
+    return simplify_controlled_rotation(operation)
+
+
+def two_non_simplified_crot(operation1, operation2):
+    r"""Check commutation for two CRot that were not simplified.
+
+    Args:
+        operation1 (pennylane.Operation): First operation.
+        operation2 (pennylane.Operation): Second operation.
+
+    Returns:
+         Bool: True if commutation, False otherwise.
+    """
+    # Two non simplified CRot
+
+    control_control = intersection(operation1.control_wires, operation2.control_wires)
+    target_target = intersection(operation1.target_wires, operation2.target_wires)
+
+    if control_control and target_target:
+        return np.all(
+            np.allclose(
+                np.matmul(operation1.matrix, operation2.matrix),
+                np.matmul(operation2.matrix, operation1.matrix),
+            )
+        )
+    elif control_control and not target_target:
+        return True
+    elif not control_control and target_target:
+        return np.all(
+            np.allclose(
+                np.matmul(
+                    qml.Rot(*operation1.data, wires=operation1.wires[1]).matrix,
+                    qml.Rot(*operation2.data, wires=operation2.wires[1]).matrix,
+                ),
+                np.matmul(
+                    qml.Rot(*operation2.data, wires=operation2.wires[1]).matrix,
+                    qml.Rot(*operation1.data, wires=operation1.wires[1]).matrix,
```

Comment:
Same here, the `.matrix` should be updated to `.get_matrix()`.

Reply:
I know I mentioned above that no function or method names should begin with `get_matrix()` 😆 This is just temporary, and we will change this before the release.